Sources/AblyChat/RoomLifecycleManager.swift

```diff
@@ -0,0 +1,297 @@
+                    while true {
+                        do {
+                            logger.log(message: "Will attempt to detach non-failed contributor \(contributor) in 1s.", level: .info)
+                            // TODO: what's the correct wait time? (https://github.com/ably/specification/pull/200#discussion_r1763799223)
+                            try await clock.sleep(timeInterval: 1)
+                            logger.log(message: "Detaching non-failed contributor \(contributor)", level: .info)
+                            try await contributor.channel.detach()
+                            break
+                        } catch {
+                            // Loop repeats
+                            logger.log(message: "Failed to detach non-failed contributor \(contributor), error \(error). Will retry.", level: .info)
+                        }
+                    }
+                }
+            }
+        }
+
+        if let firstDetachError {
+            // CHA-RL2f
+            throw firstDetachError
+        }
+
+        // CHA-RL2g
+        changeStatus(to: .detached)
+    }
+
+    /// Implementes CHA-RL3’s RELEASE operation.
+    internal func performReleaseOperation() async {
+        switch current {
+        case .released:
+            // CHA-RL3a
+            return
+        case .detached:
+            // CHA-RL3b
+            changeStatus(to: .released)
+            return
+        case .releasing, .initialized, .attached, .attaching, .detaching, .suspended, .failed:
+            break
+        }
+
+        changeStatus(to: .releasing)
+
+        // CHA-RL3d
+        for contributor in contributors {
+            detachLoop: while true {
+                let contributorState = await contributor.channel.state
+
+                // CHA-RL3e
+                guard contributorState != .failed else {
+                    logger.log(message: "Contributor \(contributor) is FAILED; skipping detach", level: .info)
+                    break
+                }
+
+                logger.log(message: "Detaching contributor \(contributor)", level: .info)
+                do {
+                    try await contributor.channel.detach()
+                    break detachLoop
+                } catch {
+                    // CHA-RL3f: Retry until detach succeeds, with a pause before each attempt
+                    logger.log(message: "Failed to detach contributor \(contributor), error \(error). Will retry in 1s.", level: .info)
+                    // TODO: Make this not trap in the case where the Task is cancelled (as part of the broader https://github.com/ably-labs/ably-chat-swift/issues/29 for handling task cancellation)
+                    // TODO: what's the correct wait time? (https://github.com/ably/specification/pull/200#discussion_r1763822207)
+                    // swiftlint:disable:next force_try
+                    try! await clock.sleep(timeInterval: 1)
```

Comment:
**suggestion (bug_risk):** Avoid force unwrapping and handle potential errors

Force unwrapping can lead to runtime crashes. Consider using optional binding or throwing functions to handle potential errors safely. This will make the code more robust and less prone to unexpected crashes.